pkg/syncthing/syncthing.go

```diff
@@ -486,68 +507,108 @@ func (s *Syncthing) GetCompletion(ctx context.Context, local bool, device string
 
 // IsHealthy returns the syncthing error or nil
 func (s *Syncthing) IsHealthy(ctx context.Context, local bool, max int) error {
-	for _, folder := range s.Folders {
-		status, err := s.GetStatus(ctx, folder, false)
-		if err != nil {
-			if err == errors.ErrBusySyncthing {
-				continue
-			}
-			return err
-		}
-		if status.PullErrors == 0 {
-			folder.Retries = 0
-			continue
+	pullErrors, err := s.GetPullErrors(ctx, local)
+	if err != nil {
+		if err == errors.ErrBusySyncthing {
+			return nil
 		}
+		return err
+	}
+	if pullErrors == 0 {
+		isHealthyRetries = 0
+		return nil
+	}
 
-		folder.Retries++
-		err = s.GetFolderErrors(ctx, folder, false)
-		if err != nil {
-			log.Infof("syncthing error in folder '%s' local=%t retry %d: %s", folder.RemotePath, local, folder.Retries, err.Error())
-		}
-		if err == errors.ErrInsufficientSpace {
-			return err
-		}
+	isHealthyRetries++
```

Comment:
Why is it global now? Probably I'm missing context, I've never worked with syncthing

Reply:
Before, for each folder in sync, we were counting the number of retries for that folder. It was a global state, but now we can use a single variable because we are checking errors on any folder in a single query.